tests/gtest/test_rcp_host_api.cpp

```diff
@@ -110,6 +111,7 @@ TEST(RcpHostApi, DeviceRoleChangesCorrectlyAfterSetThreadEnabled)
     MainloopProcessUntil(mainloop, /* aTimeoutSec */ 1, [&resultReceived]() { return resultReceived; });
```

Comment:
may verify the state is disabling before the delay?

Reply:
Added the check.